### PR TITLE
Zoltan2: small modifications for HIP

### PR DESCRIPTION
**Pull request by lucbv**
@trilinos/zoltan2 

## Motivation
Mostly changing a few memory_space to device_type, this allows the proper handling of special devices that might use HostPinnedMemory.
Also changing pre-processor guards and some CMake logic.

## Related Issues

* Closes 
* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to #9322 
* Part of 
* Composed of 

## Testing
This PR was tested on spock at ORNL using MI100 GPUs, see results below.
```
lberge@spock13:/gpfs/alpine/scratch/lberge/csc465/builds/trilinos_lucbv/packages/zoltan2> ctest -R
Test project /gpfs/alpine/scratch/lberge/csc465/builds/trilinos_lucbv/packages/zoltan2
        Start   1: Zoltan2_Machine_MPI_4
  1/153 Test   #1: Zoltan2_Machine_MPI_4 .......................................   Passed    0.91 sec
        Start   2: Zoltan2_TimerManager_MPI_4
  2/153 Test   #2: Zoltan2_TimerManager_MPI_4 ..................................   Passed   13.82 sec
        Start   3: Zoltan2_DebugManager_MPI_4
  3/153 Test   #3: Zoltan2_DebugManager_MPI_4 ..................................   Passed    0.95 sec
        Start   4: Zoltan2_Environment_MPI_4
  4/153 Test   #4: Zoltan2_Environment_MPI_4 ...................................   Passed    0.83 sec
        Start   5: Zoltan2_MetricOutputManager_MPI_4
  5/153 Test   #5: Zoltan2_MetricOutputManager_MPI_4 ...........................   Passed    0.93 sec
        Start   6: Zoltan2_Parameters_MPI_1
  6/153 Test   #6: Zoltan2_Parameters_MPI_1 ....................................   Passed    0.63 sec
        Start   7: Zoltan2_AllParameters_MPI_1
  7/153 Test   #7: Zoltan2_AllParameters_MPI_1 .................................   Passed    0.64 sec
        Start   8: Zoltan2_InputTraitsGood_MPI_1
  8/153 Test   #8: Zoltan2_InputTraitsGood_MPI_1 ...............................   Passed    0.13 sec
        Start   9: Zoltan2_InputTraitsBad_MPI_1
  9/153 Test   #9: Zoltan2_InputTraitsBad_MPI_1 ................................   Passed    0.12 sec
        Start  10: Zoltan2_BasicCoordinateInput_MPI_4
 10/153 Test  #10: Zoltan2_BasicCoordinateInput_MPI_4 ..........................   Passed    0.85 sec
        Start  11: Zoltan2_BasicIdentifierInput_MPI_4
 11/153 Test  #11: Zoltan2_BasicIdentifierInput_MPI_4 ..........................   Passed    0.82 sec
        Start  12: Zoltan2_BasicKokkosIdentifierInput_MPI_4
 12/153 Test  #12: Zoltan2_BasicKokkosIdentifierInput_MPI_4 ....................   Passed    0.82 sec
        Start  13: Zoltan2_BasicVectorInput_MPI_4
 13/153 Test  #13: Zoltan2_BasicVectorInput_MPI_4 ..............................   Passed    0.81 sec
        Start  14: Zoltan2_TpetraRowMatrixInput_MPI_4
 14/153 Test  #14: Zoltan2_TpetraRowMatrixInput_MPI_4 ..........................   Passed    0.87 sec
        Start  15: Zoltan2_TpetraRowGraphInput_MPI_4
 15/153 Test  #15: Zoltan2_TpetraRowGraphInput_MPI_4 ...........................   Passed    0.86 sec
        Start  16: Zoltan2_XpetraTraits_MPI_4
 16/153 Test  #16: Zoltan2_XpetraTraits_MPI_4 ..................................   Passed    0.96 sec
        Start  17: Zoltan2_XpetraCrsMatrixInput_MPI_4
 17/153 Test  #17: Zoltan2_XpetraCrsMatrixInput_MPI_4 ..........................   Passed    0.90 sec
        Start  18: Zoltan2_XpetraCrsGraphInput_MPI_4
 18/153 Test  #18: Zoltan2_XpetraCrsGraphInput_MPI_4 ...........................   Passed    0.88 sec
        Start  19: Zoltan2_XpetraVectorInput_MPI_4
 19/153 Test  #19: Zoltan2_XpetraVectorInput_MPI_4 .............................   Passed    0.90 sec
        Start  20: Zoltan2_XpetraMultiVectorInput_noweights_MPI_4
 20/153 Test  #20: Zoltan2_XpetraMultiVectorInput_noweights_MPI_4 ..............   Passed    0.87 sec
        Start  21: Zoltan2_XpetraMultiVectorInput_oneweight_MPI_4
 21/153 Test  #21: Zoltan2_XpetraMultiVectorInput_oneweight_MPI_4 ..............   Passed    0.86 sec
        Start  22: Zoltan2_XpetraMultiVectorInput_twoweights_MPI_4
 22/153 Test  #22: Zoltan2_XpetraMultiVectorInput_twoweights_MPI_4 .............   Passed    0.87 sec
        Start  23: Zoltan2_GraphModel_MPI_4
 23/153 Test  #23: Zoltan2_GraphModel_MPI_4 ....................................   Passed    1.04 sec
        Start  24: Zoltan2_IdentifierModel_MPI_4
 24/153 Test  #24: Zoltan2_IdentifierModel_MPI_4 ...............................   Passed    0.85 sec
        Start  25: Zoltan2_CoordinateModel_MPI_4
 25/153 Test  #25: Zoltan2_CoordinateModel_MPI_4 ...............................   Passed    0.87 sec
        Start  26: Zoltan2_PartitioningSolution_MPI_4
 26/153 Test  #26: Zoltan2_PartitioningSolution_MPI_4 ..........................   Passed    0.83 sec
        Start  27: Zoltan2_Mapping_MPI_4
 27/153 Test  #27: Zoltan2_Mapping_MPI_4 .......................................   Passed    0.83 sec
        Start  28: Zoltan2_StridedData_MPI_4
 28/153 Test  #28: Zoltan2_StridedData_MPI_4 ...................................   Passed    0.82 sec
        Start  29: Zoltan2_Metric_MPI_4
 29/153 Test  #29: Zoltan2_Metric_MPI_4 ........................................   Passed    0.88 sec
        Start  30: Zoltan2_TPLTraits_MPI_4
 30/153 Test  #30: Zoltan2_TPLTraits_MPI_4 .....................................   Passed    0.81 sec
        Start  31: Zoltan2_AlltoAll_MPI_4
 31/153 Test  #31: Zoltan2_AlltoAll_MPI_4 ......................................   Passed    0.81 sec
        Start  32: Zoltan2_componentMetrics_MPI_4
 32/153 Test  #32: Zoltan2_componentMetrics_MPI_4 ..............................   Passed    0.85 sec
        Start  33: Zoltan2_findUniqueGids_MPI_4
 33/153 Test  #33: Zoltan2_findUniqueGids_MPI_4 ................................   Passed    0.83 sec
        Start  34: Zoltan2_rcbTest_MPI_4
 34/153 Test  #34: Zoltan2_rcbTest_MPI_4 .......................................   Passed    0.92 sec
        Start  35: Zoltan2_fix2010_MPI_4
 35/153 Test  #35: Zoltan2_fix2010_MPI_4 .......................................   Passed    0.82 sec
        Start  36: Zoltan2_mj_int_coordinates_MPI_4
 36/153 Test  #36: Zoltan2_mj_int_coordinates_MPI_4 ............................   Passed    0.85 sec
        Start  37: Zoltan2_mj_backwardcompat_MPI_4
 37/153 Test  #37: Zoltan2_mj_backwardcompat_MPI_4 .............................   Passed    0.82 sec
        Start  38: Zoltan2_fix4785_MPI_4
 38/153 Test  #38: Zoltan2_fix4785_MPI_4 .......................................   Passed   45.02 sec
        Start  39: Zoltan2_mjTest_MPI_4
 39/153 Test  #39: Zoltan2_mjTest_MPI_4 ........................................   Passed    0.92 sec
        Start  40: Zoltan2_mjTest_simple_rectilinear_boxes_MPI_4
 40/153 Test  #40: Zoltan2_mjTest_simple_rectilinear_boxes_MPI_4 ...............   Passed    0.87 sec
        Start  41: Zoltan2_mjTest_geomgen_MPI_3
 41/153 Test  #41: Zoltan2_mjTest_geomgen_MPI_3 ................................   Passed    0.84 sec
        Start  42: Zoltan2_mjTest_geomgen_noholes_MPI_3
 42/153 Test  #42: Zoltan2_mjTest_geomgen_noholes_MPI_3 ........................   Passed    0.84 sec
        Start  43: Zoltan2_mjTest_geomgen_nomjparts_MPI_3
 43/153 Test  #43: Zoltan2_mjTest_geomgen_nomjparts_MPI_3 ......................   Passed    0.83 sec
        Start  44: Zoltan2_mjTest_geomgen_nomjparts_odd_MPI_3
 44/153 Test  #44: Zoltan2_mjTest_geomgen_nomjparts_odd_MPI_3 ..................   Passed    0.82 sec
        Start  45: Zoltan2_mjTest_uvmOff_MPI_4
 45/153 Test  #45: Zoltan2_mjTest_uvmOff_MPI_4 .................................   Passed    0.85 sec
        Start  46: Zoltan2_mjTest_geomgen_uvmOff_MPI_3
 46/153 Test  #46: Zoltan2_mjTest_geomgen_uvmOff_MPI_3 .........................   Passed    0.84 sec
        Start  47: Zoltan2_mjTest_geomgen_noholes_uvmOff_MPI_3
 47/153 Test  #47: Zoltan2_mjTest_geomgen_noholes_uvmOff_MPI_3 .................   Passed    0.84 sec
        Start  48: Zoltan2_mjTest_geomgen_nomjparts_uvmOff_MPI_3
 48/153 Test  #48: Zoltan2_mjTest_geomgen_nomjparts_uvmOff_MPI_3 ...............   Passed    0.82 sec
        Start  49: Zoltan2_mjTest_geomgen_nomjparts_odd_uvmOff_MPI_3
 49/153 Test  #49: Zoltan2_mjTest_geomgen_nomjparts_odd_uvmOff_MPI_3 ...........   Passed    0.84 sec
        Start  50: Zoltan2_sarmaTest_MPI_1
 50/153 Test  #50: Zoltan2_sarmaTest_MPI_1 .....................................   Passed    0.13 sec
        Start  51: Zoltan2_TaskMappingTest3_MPI_1
 51/153 Test  #51: Zoltan2_TaskMappingTest3_MPI_1 ..............................   Passed    2.33 sec
        Start  52: Zoltan2_TaskMappingTest_MPI_4
 52/153 Test  #52: Zoltan2_TaskMappingTest_MPI_4 ...............................   Passed    0.90 sec
        Start  53: Zoltan2_TaskMappingProblemTest_MPI_4
 53/153 Test  #53: Zoltan2_TaskMappingProblemTest_MPI_4 ........................   Passed    0.92 sec
        Start  54: Zoltan2_blockTest_MPI_4
 54/153 Test  #54: Zoltan2_blockTest_MPI_4 .....................................   Passed    0.82 sec
        Start  55: Zoltan2_Partitioning1_MPI_4
 55/153 Test  #55: Zoltan2_Partitioning1_MPI_4 .................................   Passed    0.88 sec
        Start  56: Zoltan2_Partitioning1_OneProc_MPI_4
 56/153 Test  #56: Zoltan2_Partitioning1_OneProc_MPI_4 .........................   Passed    0.86 sec
        Start  57: Zoltan2_Partitioning1_VWeights_MPI_4
 57/153 Test  #57: Zoltan2_Partitioning1_VWeights_MPI_4 ........................   Passed    0.86 sec
        Start  58: Zoltan2_Partitioning1_OneProc_VWeights_MPI_4
 58/153 Test  #58: Zoltan2_Partitioning1_OneProc_VWeights_MPI_4 ................   Passed    0.86 sec
        Start  59: Zoltan2_Partitioning1_EWeights_MPI_4
 59/153 Test  #59: Zoltan2_Partitioning1_EWeights_MPI_4 ........................   Passed    0.86 sec
        Start  60: Zoltan2_Partitioning1_OneProc_EWeights_MPI_4
 60/153 Test  #60: Zoltan2_Partitioning1_OneProc_EWeights_MPI_4 ................   Passed    0.88 sec
        Start  61: Zoltan2_Partitioning1_ParMETIS_MPI_4
 61/153 Test  #61: Zoltan2_Partitioning1_ParMETIS_MPI_4 ........................   Passed    0.85 sec
        Start  62: Zoltan2_Partitioning1_ParMETIS_OneProc_MPI_4
 62/153 Test  #62: Zoltan2_Partitioning1_ParMETIS_OneProc_MPI_4 ................   Passed    0.86 sec
        Start  63: Zoltan2_Partitioning1_ParMETIS_VWeights_MPI_4
 63/153 Test  #63: Zoltan2_Partitioning1_ParMETIS_VWeights_MPI_4 ...............   Passed    0.87 sec
        Start  64: Zoltan2_Partitioning1_ParMETIS_OneProc_VWeights_MPI_4
 64/153 Test  #64: Zoltan2_Partitioning1_ParMETIS_OneProc_VWeights_MPI_4 .......   Passed    0.86 sec
        Start  65: Zoltan2_Partitioning1_ParMETIS_EWeights_MPI_4
 65/153 Test  #65: Zoltan2_Partitioning1_ParMETIS_EWeights_MPI_4 ...............   Passed    0.86 sec
        Start  66: Zoltan2_Partitioning1_ParMETIS_OneProc_EWeights_MPI_4
 66/153 Test  #66: Zoltan2_Partitioning1_ParMETIS_OneProc_EWeights_MPI_4 .......   Passed    0.86 sec
        Start  67: Zoltan2_Partitioning1_Quotient_MPI_4
 67/153 Test  #67: Zoltan2_Partitioning1_Quotient_MPI_4 ........................   Passed    0.87 sec
        Start  68: Zoltan2_Partitioning1_Quotient_OneProc_MPI_4
 68/153 Test  #68: Zoltan2_Partitioning1_Quotient_OneProc_MPI_4 ................   Passed    0.87 sec
        Start  69: Zoltan2_Partitioning1_Quotient_Threshold1_MPI_4
 69/153 Test  #69: Zoltan2_Partitioning1_Quotient_Threshold1_MPI_4 .............   Passed    0.86 sec
        Start  70: Zoltan2_Partitioning1_Quotient_Threshold2_MPI_4
 70/153 Test  #70: Zoltan2_Partitioning1_Quotient_Threshold2_MPI_4 .............   Passed    0.86 sec
        Start  71: Zoltan2_Partitioning1_Quotient_OneProc_Threshold1_MPI_4
 71/153 Test  #71: Zoltan2_Partitioning1_Quotient_OneProc_Threshold1_MPI_4 .....   Passed    0.88 sec
        Start  72: Zoltan2_Partitioning1_Quotient_OneProc_Threshold2_MPI_4
 72/153 Test  #72: Zoltan2_Partitioning1_Quotient_OneProc_Threshold2_MPI_4 .....   Passed    0.87 sec
        Start  73: Zoltan2_PartitioningTree_MPI_4
 73/153 Test  #73: Zoltan2_PartitioningTree_MPI_4 ..............................   Passed    0.87 sec
        Start  74: Zoltan2_Ordering1_MPI_4
 74/153 Test  #74: Zoltan2_Ordering1_MPI_4 .....................................   Passed    0.85 sec
        Start  75: Zoltan2_Ordering1_Ashear_MPI_4
 75/153 Test  #75: Zoltan2_Ordering1_Ashear_MPI_4 ..............................   Passed    0.85 sec
        Start  76: Zoltan2_Ordering1_natural_MPI_4
 76/153 Test  #76: Zoltan2_Ordering1_natural_MPI_4 .............................   Passed    0.87 sec
        Start  77: Zoltan2_Ordering1_random_MPI_4
 77/153 Test  #77: Zoltan2_Ordering1_random_MPI_4 ..............................   Passed    0.84 sec
        Start  78: Zoltan2_Ordering1_rcm_MPI_4
 78/153 Test  #78: Zoltan2_Ordering1_rcm_MPI_4 .................................   Passed    0.85 sec
        Start  79: Zoltan2_Ordering1_sorteddegree_MPI_4
 79/153 Test  #79: Zoltan2_Ordering1_sorteddegree_MPI_4 ........................   Passed    0.83 sec
        Start  80: Zoltan2_OrderingAMD_MPI_4
 80/153 Test  #80: Zoltan2_OrderingAMD_MPI_4 ...................................   Passed    0.86 sec
        Start  81: Zoltan2_TpetraCrsColorer_simple_sym_MPI_4
 81/153 Test  #81: Zoltan2_TpetraCrsColorer_simple_sym_MPI_4 ...................   Passed    0.92 sec
        Start  82: Zoltan2_TpetraCrsColorer_simple_nodistrib_sym_MPI_4
 82/153 Test  #82: Zoltan2_TpetraCrsColorer_simple_nodistrib_sym_MPI_4 .........   Passed    0.90 sec
        Start  83: Zoltan2_TpetraCrsColorer_simple_MPI_4
 83/153 Test  #83: Zoltan2_TpetraCrsColorer_simple_MPI_4 .......................   Passed    0.95 sec
        Start  84: Zoltan2_TpetraCrsColorer_simple_nodistrib_MPI_4
 84/153 Test  #84: Zoltan2_TpetraCrsColorer_simple_nodistrib_MPI_4 .............   Passed    0.95 sec
        Start  85: Zoltan2_TpetraCrsColorer_west0067_MPI_4
 85/153 Test  #85: Zoltan2_TpetraCrsColorer_west0067_MPI_4 .....................   Passed    0.95 sec
        Start  86: Zoltan2_TpetraCrsColorer_west0067_nodistrib_MPI_4
 86/153 Test  #86: Zoltan2_TpetraCrsColorer_west0067_nodistrib_MPI_4 ...........   Passed    0.96 sec
        Start  87: Zoltan2_TpetraCrsColorer_galeri1_MPI_4
 87/153 Test  #87: Zoltan2_TpetraCrsColorer_galeri1_MPI_4 ......................   Passed    0.93 sec
        Start  88: Zoltan2_TpetraCrsColorer_galeri2_MPI_4
 88/153 Test  #88: Zoltan2_TpetraCrsColorer_galeri2_MPI_4 ......................   Passed    0.99 sec
        Start  89: Zoltan2_TpetraCrsColorer_galeri2_sym_MPI_4
 89/153 Test  #89: Zoltan2_TpetraCrsColorer_galeri2_sym_MPI_4 ..................   Passed    0.98 sec
        Start  90: Zoltan2_TpetraCrsColorer_troublecauser_MPI_4
 90/153 Test  #90: Zoltan2_TpetraCrsColorer_troublecauser_MPI_4 ................   Passed    1.01 sec
        Start  91: Zoltan2_Coloring1_MPI_4
 91/153 Test  #91: Zoltan2_Coloring1_MPI_4 .....................................   Passed    0.86 sec
        Start  92: Zoltan2_Coloring1_simple_MPI_4
 92/153 Test  #92: Zoltan2_Coloring1_simple_MPI_4 ..............................   Passed    0.84 sec
        Start  93: Zoltan2_Coloring1_commanchedual_MPI_4
 93/153 Test  #93: Zoltan2_Coloring1_commanchedual_MPI_4 .......................   Passed    0.89 sec
        Start  94: Zoltan2_Coloring1_simple_D1_MPI_4
 94/153 Test  #94: Zoltan2_Coloring1_simple_D1_MPI_4 ...........................   Passed    0.85 sec
        Start  95: Zoltan2_Coloring1_simple_D1_recolor_degrees_MPI_4
 95/153 Test  #95: Zoltan2_Coloring1_simple_D1_recolor_degrees_MPI_4 ...........   Passed    0.85 sec
        Start  96: Zoltan2_Coloring1_simple_D1_serial_threshold_25_MPI_4
 96/153 Test  #96: Zoltan2_Coloring1_simple_D1_serial_threshold_25_MPI_4 .......   Passed    0.86 sec
        Start  97: Zoltan2_Coloring1_commanchedual_D1_MPI_4
 97/153 Test  #97: Zoltan2_Coloring1_commanchedual_D1_MPI_4 ....................   Passed    0.89 sec
        Start  98: Zoltan2_Coloring1_noedges_D1_MPI_4
 98/153 Test  #98: Zoltan2_Coloring1_noedges_D1_MPI_4 ..........................   Passed    0.87 sec
        Start  99: Zoltan2_Coloring1_complete_D1_MPI_4
 99/153 Test  #99: Zoltan2_Coloring1_complete_D1_MPI_4 .........................   Passed    0.85 sec
        Start 100: Zoltan2_Coloring1_simple_D1-2GL_MPI_4
100/153 Test #100: Zoltan2_Coloring1_simple_D1-2GL_MPI_4 .......................   Passed    0.86 sec
        Start 101: Zoltan2_Coloring1_simple_D1-2GL_recolor_degrees_MPI_4
101/153 Test #101: Zoltan2_Coloring1_simple_D1-2GL_recolor_degrees_MPI_4 .......   Passed    0.87 sec
        Start 102: Zoltan2_Coloring1_simple_D1-2GL_serial_threshold_25_MPI_4
102/153 Test #102: Zoltan2_Coloring1_simple_D1-2GL_serial_threshold_25_MPI_4 ...   Passed    0.86 sec
        Start 103: Zoltan2_Coloring1_commanchedual_D1-2GL_MPI_4
103/153 Test #103: Zoltan2_Coloring1_commanchedual_D1-2GL_MPI_4 ................   Passed    0.89 sec
        Start 104: Zoltan2_Coloring1_noedges_D1-2GL_MPI_4
104/153 Test #104: Zoltan2_Coloring1_noedges_D1-2GL_MPI_4 ......................   Passed    0.86 sec
        Start 105: Zoltan2_Coloring1_complete_D1-2GL_MPI_4
105/153 Test #105: Zoltan2_Coloring1_complete_D1-2GL_MPI_4 .....................   Passed    0.86 sec
        Start 106: Zoltan2_Coloring1_simple_D2_MPI_4
106/153 Test #106: Zoltan2_Coloring1_simple_D2_MPI_4 ...........................   Passed    0.89 sec
        Start 107: Zoltan2_Coloring1_simple_D2_recolor_degrees_MPI_4
107/153 Test #107: Zoltan2_Coloring1_simple_D2_recolor_degrees_MPI_4 ...........   Passed    0.91 sec
        Start 108: Zoltan2_Coloring1_simple_D2_serial_threshold_25_MPI_4
108/153 Test #108: Zoltan2_Coloring1_simple_D2_serial_threshold_25_MPI_4 .......   Passed    0.88 sec
        Start 109: Zoltan2_Coloring1_commanchedual_D2_MPI_4
109/153 Test #109: Zoltan2_Coloring1_commanchedual_D2_MPI_4 ....................   Passed    0.95 sec
        Start 110: Zoltan2_Coloring1_noedges_D2_MPI_4
110/153 Test #110: Zoltan2_Coloring1_noedges_D2_MPI_4 ..........................   Passed    0.87 sec
        Start 111: Zoltan2_Coloring1_complete_D2_MPI_4
111/153 Test #111: Zoltan2_Coloring1_complete_D2_MPI_4 .........................   Passed    0.90 sec
        Start 112: Zoltan2_Coloring1_simple_PD2_MPI_4
112/153 Test #112: Zoltan2_Coloring1_simple_PD2_MPI_4 ..........................   Passed    0.90 sec
        Start 113: Zoltan2_Coloring1_simple_PD2_recolor_degrees_MPI_4
113/153 Test #113: Zoltan2_Coloring1_simple_PD2_recolor_degrees_MPI_4 ..........   Passed    0.89 sec
        Start 114: Zoltan2_Coloring1_simple_PD2_serial_threshold_25_MPI_4
114/153 Test #114: Zoltan2_Coloring1_simple_PD2_serial_threshold_25_MPI_4 ......   Passed    0.89 sec
        Start 115: Zoltan2_Coloring1_commanchedual_PD2_MPI_4
115/153 Test #115: Zoltan2_Coloring1_commanchedual_PD2_MPI_4 ...................   Passed    0.94 sec
        Start 116: Zoltan2_Coloring1_noedges_PD2_MPI_4
116/153 Test #116: Zoltan2_Coloring1_noedges_PD2_MPI_4 .........................   Passed    0.87 sec
        Start 117: Zoltan2_Coloring1_complete_PD2_MPI_4
117/153 Test #117: Zoltan2_Coloring1_complete_PD2_MPI_4 ........................   Passed    0.90 sec
        Start 118: Zoltan2_rcbPerformanceZ1_MPI_4
118/153 Test #118: Zoltan2_rcbPerformanceZ1_MPI_4 ..............................   Passed    0.85 sec
        Start 119: Zoltan2_testfail1_MPI_4
119/153 Test #119: Zoltan2_testfail1_MPI_4 .....................................   Passed    0.23 sec
        Start 120: Zoltan2_testfail2_MPI_4
120/153 Test #120: Zoltan2_testfail2_MPI_4 .....................................   Passed    0.23 sec
        Start 121: Zoltan2_ddirectoryTest_MPI_4
121/153 Test #121: Zoltan2_ddirectoryTest_MPI_4 ................................   Passed    0.84 sec
        Start 122: Zoltan2_teuchosCommTest_MPI_4
122/153 Test #122: Zoltan2_teuchosCommTest_MPI_4 ...............................   Passed    0.38 sec
        Start 123: Zoltan2_absDefineTest_MPI_1
123/153 Test #123: Zoltan2_absDefineTest_MPI_1 .................................   Passed    0.10 sec
        Start 124: Zoltan2_tpetraOneToOneMapTest_MPI_4
124/153 Test #124: Zoltan2_tpetraOneToOneMapTest_MPI_4 .........................   Passed    0.81 sec
        Start 125: Zoltan2_tpetraVecWithCopiesTest_MPI_4
125/153 Test #125: Zoltan2_tpetraVecWithCopiesTest_MPI_4 .......................   Passed    0.85 sec
        Start 126: Zoltan2_geometricTest_MPI_4
126/153 Test #126: Zoltan2_geometricTest_MPI_4 .................................   Passed    0.86 sec
        Start 127: Zoltan2_test578_MPI_4
127/153 Test #127: Zoltan2_test578_MPI_4 .......................................   Passed    0.85 sec
        Start 128: Zoltan2_geometricVwgtTest_MPI_4
128/153 Test #128: Zoltan2_geometricVwgtTest_MPI_4 .............................   Passed    0.88 sec
        Start 129: Zoltan2_geometricVwgt2Test_MPI_4
129/153 Test #129: Zoltan2_geometricVwgt2Test_MPI_4 ............................   Passed    0.89 sec
        Start 130: Zoltan2_multijaggedVwgt2Test_MPI_4
130/153 Test #130: Zoltan2_multijaggedVwgt2Test_MPI_4 ..........................   Passed    0.89 sec
        Start 131: Zoltan2_coordinateDriverTest_MPI_4
131/153 Test #131: Zoltan2_coordinateDriverTest_MPI_4 ..........................   Passed    0.92 sec
        Start 132: Zoltan2_rcbDriverTest_MPI_4
132/153 Test #132: Zoltan2_rcbDriverTest_MPI_4 .................................   Passed    0.89 sec
        Start 133: Zoltan2_mjDriverTest_MPI_3
133/153 Test #133: Zoltan2_mjDriverTest_MPI_3 ..................................   Passed    0.81 sec
        Start 134: Zoltan2_mjTestPremigrate_MPI_4
134/153 Test #134: Zoltan2_mjTestPremigrate_MPI_4 ..............................   Passed    0.86 sec
        Start 135: Zoltan2_zoltanWithGraphAdapter_MPI_4
135/153 Test #135: Zoltan2_zoltanWithGraphAdapter_MPI_4 ........................   Passed    0.87 sec
        Start 136: Zoltan2_zoltanWithGraphAdapterVwgt_MPI_4
136/153 Test #136: Zoltan2_zoltanWithGraphAdapterVwgt_MPI_4 ....................   Passed    0.88 sec
        Start 137: Zoltan2_zoltanWithGraphAdapterEwgt_MPI_4
137/153 Test #137: Zoltan2_zoltanWithGraphAdapterEwgt_MPI_4 ....................   Passed    0.86 sec
        Start 138: Zoltan2_chacoGraphMetricsTest_MPI_4
138/153 Test #138: Zoltan2_chacoGraphMetricsTest_MPI_4 .........................   Passed    0.88 sec
        Start 139: Zoltan2_chacoGraphMetricsTestNoDistribute_MPI_4
139/153 Test #139: Zoltan2_chacoGraphMetricsTestNoDistribute_MPI_4 .............   Passed    0.84 sec
        Start 140: Zoltan2_chacoGraphMetricsTestNoGraph_MPI_4
140/153 Test #140: Zoltan2_chacoGraphMetricsTestNoGraph_MPI_4 ..................   Passed    0.86 sec
        Start 141: Zoltan2_chacoGraphMetricsTestNoGraphNoDistribute_MPI_4
141/153 Test #141: Zoltan2_chacoGraphMetricsTestNoGraphNoDistribute_MPI_4 ......   Passed    0.84 sec
        Start 142: Zoltan2_coffeyMJ_MPI_1
142/153 Test #142: Zoltan2_coffeyMJ_MPI_1 ......................................   Passed    0.67 sec
        Start 143: Zoltan2_glazeRCB_MPI_2
143/153 Test #143: Zoltan2_glazeRCB_MPI_2 ......................................   Passed    0.83 sec
        Start 144: Zoltan2_glaze2RCB_MPI_2
144/153 Test #144: Zoltan2_glaze2RCB_MPI_2 .....................................   Passed    0.82 sec
        Start 145: Zoltan2_rectilinearRCB_left_MPI_2
145/153 Test #145: Zoltan2_rectilinearRCB_left_MPI_2 ...........................   Passed    0.81 sec
        Start 146: Zoltan2_rectilinearRCB_right_MPI_2
146/153 Test #146: Zoltan2_rectilinearRCB_right_MPI_2 ..........................   Passed    0.79 sec
        Start 147: Zoltan2_kuberry_MPI_4
147/153 Test #147: Zoltan2_kuberry_MPI_4 .......................................   Passed    0.86 sec
        Start 148: Zoltan2_directoryTest_Kokkos_MPI_4
148/153 Test #148: Zoltan2_directoryTest_Kokkos_MPI_4 ..........................   Passed    0.90 sec
        Start 149: Zoltan2_directoryTest_KokkosSimple_MPI_4
149/153 Test #149: Zoltan2_directoryTest_KokkosSimple_MPI_4 ....................   Passed    0.82 sec
        Start 150: Zoltan2_directoryTest_findUniqueGids.cpp_MPI_4
150/153 Test #150: Zoltan2_directoryTest_findUniqueGids.cpp_MPI_4 ..............   Passed    0.82 sec
        Start 151: Zoltan2_block_MPI_4
151/153 Test #151: Zoltan2_block_MPI_4 .........................................   Passed    0.36 sec
        Start 152: Zoltan2_kokkosBlock_MPI_4
152/153 Test #152: Zoltan2_kokkosBlock_MPI_4 ...................................   Passed    0.82 sec
        Start 153: Zoltan2_rcb_C_MPI_4
153/153 Test #153: Zoltan2_rcb_C_MPI_4 .........................................   Passed    0.83 sec

100% tests passed, 0 tests failed out of 153

Label Time Summary:
Zoltan2    = 714.10 sec*proc (153 tests)

Total Test time (real) = 185.68 sec
```

### CODE REVIEW

File: packages/zoltan2/core/src/algorithms/color/Zoltan2_AlgHybridD1.hpp
Comment (kddevin) [inline packages/zoltan2/core/src/algorithms/color/Zoltan2_AlgHybridD1.hpp:777]: All other calls to getLocalView were changed to take a device_type instead of a memory_space.  Should this call change as well?

File: packages/zoltan2/test/core/partition/MultiJaggedTest.cpp
Comment (jhux2) [inline packages/zoltan2/test/core/partition/MultiJaggedTest.cpp:1425]: Should this be an `elif`, like above?

Comment (lucbv) [reply packages/zoltan2/test/core/partition/MultiJaggedTest.cpp:1425]: Not necessarily, the `elif` was necessary in Zoltan2_TaskMapping.hpp but since there is no `else` close here it does not need to be.

**Comment by lucbv** [comment]
@jhux2 @kddevin 
I think this is ready for another look, sorry it took me a while to get back to it.
The auto-tester seems to be happy with it so we might be able to merge this : )

**Review by jhux2 — APPROVED** [review]
@lucbv lgtm, but please wait for @kddevin to review before merging.

**Comment by lucbv** [comment]
@jhux2 there was an automerge label on...
@kddevin if you can still give this a look, I don't mind reverting if there is an issue!

**Review by kddevin — COMMENTED** [review]
LGTM.  Thanks, Luc.

**Pull request merged**
